sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubOptions.cs

```diff
@@ -273,6 +266,42 @@ internal EventProcessorHost GetEventProcessorHost(IConfiguration config, string
             throw new InvalidOperationException("No event hub receiver named " + eventHubName);
         }
 
+        internal IEventHubConsumerClient GetEventHubConsumerClient(string eventHubName, string consumerGroup)
+        {
+            ReceiverCreds creds;
+            if (this._receiverCreds.TryGetValue(eventHubName, out creds))
+            {
+                consumerGroup ??= EventHubConsumerClient.DefaultConsumerGroupName;
+
+                // Use blob prefix support available in EPH starting in 2.2.6
+                return new EventHubConsumerClientImpl(new EventHubConsumerClient(
```

Comment:
If you're using this just to query the partition properties and available partitions,  why not use one of your `EventHubProducerClients`?    Both can perform those operations and you'll avoid having another connection/link open to the service?

Reply:
We only create a producer client when there is an output binding so there isn't always an existing `EventHubProducerClient` hanging around. I'll keep this in mind for future improvements.